roxctl/declarativeconfig/create/notifier.go

```diff
@@ -0,0 +1,224 @@
+		Short: "Create a declarative configuration for a generic notifier",
+	}
+	genericFlags := cmd.Flags()
+	n.gc = &declarativeconfig.GenericConfig{}
+	genericFlags.BoolVar(&n.gc.AuditLoggingEnabled, "audit-logging", false, "Audit logging enabled")
+	genericFlags.StringVar(&n.gc.Endpoint, "webhook-endpoint", "", "Webhook endpoint URL")
+	genericFlags.StringVar(&n.gc.Username, "webhook-username", "", "Username for the endpoint basic authentication")
+	genericFlags.StringVar(&n.gc.Password, "webhook-password", "", "Password for the endpoint basic authentication")
+	genericFlags.StringVar(&n.gc.CACertPEM, "webhook-cacert-file", "", "Endpoint CA certificate file name (PEM format)")
+	genericFlags.StringToStringVar(&n.gcHeaders, "headers", nil, "Headers (comma separated key=value pairs)")
+	genericFlags.StringToStringVar(&n.gcExtraFields, "extra-fields", nil, "Extra fields (comma separated key=value pairs)")
+	genericFlags.BoolVar(&n.gc.SkipTLSVerify, "webhook-insecure-skip-tls-verify", false, "Skip TLS verification")
+	n.genericFlagSet = genericFlags
+
+	cmd.Flags().AddFlagSet(genericFlags)
+
+	utils.Must(cmd.MarkFlagFilename("webhook-cacert-file"))
+	utils.Must(cmd.MarkFlagRequired("webhook-endpoint"))
+	return cmd
+}
+
+func (n *notifierCmd) splunkCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "splunk",
+		Args:  cobra.NoArgs,
+		RunE:  n.runE(),
+		Short: "Create a declarative configuration for a Splunk notifier",
+	}
+
+	splunkFlags := cmd.Flags()
+	n.sc = &declarativeconfig.SplunkConfig{}
+	splunkFlags.BoolVar(&n.sc.AuditLoggingEnabled, "audit-logging", false, "Audit logging enabled")
+	splunkFlags.StringVar(&n.sc.HTTPToken, "splunk-token", "", "Splunk HTTP token")
+	splunkFlags.StringVar(&n.sc.HTTPEndpoint, "splunk-endpoint", "", "Splunk HTTP endpoint")
+	splunkFlags.BoolVar(&n.sc.Insecure, "splunk-insecure-skip-tls-verify", false, "Insecure connection to Splunk")
+	splunkFlags.Int64Var(&n.sc.Truncate, "truncate", 0, "Splunk truncate limit")
+	splunkFlags.StringToStringVar(&n.scSourceTypes, "source-types", nil, "Splunk source types (comma separated key=value pairs)")
+	n.splunkFlagSet = splunkFlags
+
+	cmd.Flags().AddFlagSet(splunkFlags)
+
+	return cmd
+}
+
+type notifierCmd struct {
+	notifier *declarativeconfig.Notifier
+
+	env       environment.Environment
+	configMap string
+	secret    string
+	namespace string
+
+	genericFlagSet *pflag.FlagSet
+	gc             *declarativeconfig.GenericConfig
+	gcHeaders      map[string]string
+	gcExtraFields  map[string]string
+
+	splunkFlagSet *pflag.FlagSet
+	sc            *declarativeconfig.SplunkConfig
+	scSourceTypes map[string]string
+}
+
+func (n *notifierCmd) runE() func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := n.construct(cmd); err != nil {
+			return err
+		}
+		if err := n.validate(); err != nil {
+			return err
+		}
+		return n.printYAML()
+	}
+}
+
+func anyFlagChanged(fs *pflag.FlagSet) bool {
```

Comment:
nit: could make this a part of `roxctl/common/flags` but not strictly required IMO.

Reply:
Let's do it with a refactoring PR, when there are more users of this feature.